### ISSUE TITLE
If the player traps themselves in a visible trap, validator releases them

### ISSUE THREAD

**Issue by Labrasones** [NikVidya/Lair-Thief#11]
So, this one is tricky. Technically this is a bug, but the alternative is problematic as well.

If there is a set of rocks that will trap the player, and they are visible on the screen, the path validation step will remove a visible rock if the player traps themselves there.

i.e.
```
1) Player Moves
[ ][x][ ]
[x][ ][x]
[ ][P][ ]

2) Player Moves
[ ][x][ ]
[x][P][x]
[ ][ ][ ]

2.5) After path validation:
[ ][ ][ ]
[x][P][x]
[ ][ ][ ]
The rock above the player has been removed, within sight of the player.
```

Now, I'm not sure what to do about this. On the one hand, changing the board visibly will lead the player to believe they don't need to worry about where they are going, since they can never get stuck. However, should we allow the player to intentionally trap themselves like this?

Should this be removed? Or left in?

**Comment by soulistpvp** [comment]
Wouldn't that make our rock breaking power-up useless if we left that in?

**Comment by Labrasones** [comment]
Not necessarily. You would still be able to use that power to break through walls that don't trap you. i.e. horizontal walls
